GVRf/Framework/jni/objects/components/render_data_jni.cpp

```diff
@@ -108,152 +112,158 @@ Java_org_gearvrf_NativeRenderData_setDrawMode(
 
 JNIEXPORT jlong JNICALL
 Java_org_gearvrf_NativeRenderData_ctor(JNIEnv * env,
-        jobject obj) {
-    return reinterpret_cast<jlong>(new RenderData());
+    jobject obj) {
+return reinterpret_cast<jlong>(new RenderData());
 }
 
 JNIEXPORT void JNICALL
 Java_org_gearvrf_NativeRenderData_setMesh(JNIEnv * env,
-        jobject obj, jlong jrender_data, jlong jmesh) {
-    RenderData* render_data = reinterpret_cast<RenderData*>(jrender_data);
-    Mesh* mesh = reinterpret_cast<Mesh*>(jmesh);
-    render_data->set_mesh(mesh);
+    jobject obj, jlong jrender_data, jlong jmesh) {
+RenderData* render_data = reinterpret_cast<RenderData*>(jrender_data);
+Mesh* mesh = reinterpret_cast<Mesh*>(jmesh);
+render_data->set_mesh(mesh);
 }
 
 JNIEXPORT void JNICALL
 Java_org_gearvrf_NativeRenderData_setMaterial(JNIEnv * env,
-        jobject obj, jlong jrender_data, jlong jmaterial) {
-    RenderData* render_data = reinterpret_cast<RenderData*>(jrender_data);
-    Material* material = reinterpret_cast<Material*>(jmaterial);
-    render_data->set_material(material);
+    jobject obj, jlong jrender_data, jlong jmaterial) {
+RenderData* render_data = reinterpret_cast<RenderData*>(jrender_data);
+Material* material = reinterpret_cast<Material*>(jmaterial);
+render_data->set_material(material);
+}
+
+JNIEXPORT void JNICALL
+Java_org_gearvrf_NativeRenderData_setLight(JNIEnv * env,
+    jobject obj, jlong jrender_data, jlong jlight) {
+RenderData* render_data = reinterpret_cast<RenderData*>(jrender_data);
+Light* light = reinterpret_cast<Light*>(jlight);
+render_data->set_light(light);
 }
 
 JNIEXPORT jint JNICALL
 Java_org_gearvrf_NativeRenderData_getRenderMask(JNIEnv * env,
-        jobject obj, jlong jrender_data) {
-    RenderData* render_data = reinterpret_cast<RenderData*>(jrender_data);
-    return render_data->render_mask();
+    jobject obj, jlong jrender_data) {
+RenderData* render_data = reinterpret_cast<RenderData*>(jrender_data);
+return render_data->render_mask();
```

Comment:
The formatting here looks weird to me. It seems that it is because your eclipse can't parse codes well if there shows some syntax errors because of problematic CDT configuration.